src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -390,6 +390,121 @@ BOOST_AUTO_TEST_CASE(SimpleLinearLayerTest)
   BOOST_REQUIRE_EQUAL(arma::accu(delta), 0);
 }
 
+/**
+ * Linear module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(LinearLayerWeightInitializationTest)
+{
+  size_t inSize = 10, outSize = 4;
+  Linear<>* linear = new Linear<>(inSize, outSize);
+  linear->Reset();
+  RandomInitialization().Initialize(linear->Weight(), outSize, inSize);
+  linear->Bias().ones();
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(linear->Weight().begin(),
+      linear->Weight().end(), linear->Parameters().begin(),
+      linear->Parameters().end() - outSize);
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(linear->Bias().begin(),
+      linear->Bias().end(), linear->Parameters().begin() + inSize * outSize,
+      linear->Parameters().end());
+
+  BOOST_REQUIRE_EQUAL(linear->Weight().n_rows, outSize);
+  BOOST_REQUIRE_EQUAL(linear->Weight().n_cols, inSize);
+  BOOST_REQUIRE_EQUAL(linear->Bias().n_rows, outSize);
```

Comment:
Do we need to check `inear->Bias().n_cols == 1` as well? I am not sure it's really needed, just in case.

Reply:
Yup! I will add that.